source_middleware.go

```diff
@@ -127,46 +129,24 @@ func (c SourceWithSchemaExtractionConfig) parameters() config.Parameters {
 			},
 		},
 		configSourceSchemaExtractionPayloadEncode: {
-			Default:     strconv.FormatBool(*c.PayloadEncode),
+			Default:     strconv.FormatBool(*c.PayloadEnable),
 			Type:        config.ParameterTypeBool,
 			Description: "Whether to extract and encode the record payload with a schema.",
 		},
 		configSourceSchemaExtractionPayloadSubject: {
-			Default: func() string {
-				if c.PayloadSubject != nil {
-					return *c.PayloadSubject
-				}
-				return ""
-			}(),
-			Type: config.ParameterTypeString,
-			Description: func() string {
-				desc := "The subject of the payload schema."
-				if c.PayloadSubject == nil {
-					desc += ` Defaults to the connector ID with a ".payload" postfix.`
-				}
-				return desc
-			}(),
```

Comment:
I removed this because now the PayloadSubject will always be set (in Wrap()).

From that point of view, we probably need it as a pointer, but it might be nice leaving it as such, so that it's consistent with other fields.